metadata-api/src/main/java/com/hashmap/haf/metadata/config/actors/MetadataConfigActor.java

```diff
@@ -0,0 +1,108 @@
+import scala.concurrent.duration.Duration;
+
+@Slf4j
+public class MetadataConfigActor extends AbstractActor {
+
+    private MetadataConfig metadataConfig;
+    final Map<Integer, ActorRef> metadataQueryIdToActor = new HashMap<>();
+    final Map<ActorRef, Integer> actorToMetadataQueryId = new HashMap<>();
+
+    static public Props props() {
+        return Props.create(MetadataConfigActor.class).withDispatcher(getMetadataDispatcher());
+    }
+
+    private SupervisorStrategy strategy = new OneForOneStrategy(3, Duration.create(3, TimeUnit.SECONDS),
+            DeciderBuilder.match(Exception.class, e -> {
+                log.info("Exception {}", e.getMessage());
+                return akka.actor.SupervisorStrategy.restart();
+            })
+                    .matchAny(o -> akka.actor.SupervisorStrategy.escalate())
+                    .build());
+
+    @Override
+    public SupervisorStrategy supervisorStrategy() {
+        return strategy;
+    }
+
+    private  void processMetadataConfigMsg(MetadataMessage message) {
+        metadataConfig = message.getMetadataConfig();
+        if (message.getMessageType() == MessageType.CREATE) {
+            log.debug("Message Create metadataConfig actor");
+        } else if (message.getMessageType() == MessageType.UPDATE) {
+            log.debug("Updating metadataConfig actors for {}", metadataConfig.getId());
+            metadataConfig = message.getMetadataConfig();
+        } else if (message.getMessageType() == MessageType.DELETE) {
+            log.debug("Deleting metadataConfig actors for {}",  metadataConfig.getId());
+            context().stop(self());
+        }
+    }
+
+    private void processQueryMsg(QueryMessage message) {
+        metadataConfig = message.getMetadataConfig();
+        String query = message.getQuery();
+        ActorRef metadataQueryActor = metadataQueryIdToActor.get(query.hashCode());
+        if(metadataQueryActor != null) {
+            log.debug("Found metadataQuery actors for MetadataQueryId : {}", query.hashCode());
+            metadataQueryActor.tell(message, ActorRef.noSender());
+        } else {
+            log.debug("Creating metadataQuery actors for MetadataQueryId : {}", query.hashCode());
+            createMetadataQueryActor(message, metadataConfig);
+        }
+    }
+
+    private void createMetadataQueryActor(QueryMessage message, MetadataConfig metadataConfig) {
+        ActorRef metadataQueryActor = getContext().actorOf(MetadataQueryActor.props(metadataConfig, message.getQuery()), Integer.toString(message.getQuery().hashCode()));
+        getContext().watch(metadataQueryActor);
+        metadataQueryIdToActor.put(message.getQuery().hashCode(),metadataQueryActor);
+        actorToMetadataQueryId.put(metadataQueryActor,message.getQuery().hashCode());
+        metadataQueryActor.tell(message, ActorRef.noSender());
+    }
+
+    private void processMessage(Object message) {
+        if (message instanceof TestConnectionMsg) {
```

Comment:
I believe you removed Test connection functionality from actor and moved to Service. If that's the case remove this